File: typescript/resistor-color-duo/resistor-color-duo.ts

enum ResistorValues {
  black = 0,
  brown = 1,
  red = 2,
  orange = 3,
  yellow = 4,
  green = 5,
  blue = 6,
  violet = 7,
  grey = 8,
  white = 9,
}

type Color = keyof typeof ResistorValues;

export function decodedValue(colors: Color[]): number {
  if (!colors.length || colors.some(r => !Object.prototype.hasOwnProperty.bind(ResistorValues, r))) {
    throw new Error("error color")
  }
  let value = ResistorValues[colors[0]]
  if (colors[1]) {
    value = value * 10 + ResistorValues[colors[1]]
  }
  return value
}

File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.HashMap;
import java.util.Map;

class ResistorColorDuo {

    private final Map<String, Integer> COLOR_CODES = new HashMap<>() {
        {
            put("black", 0);
            put("brown", 1);
            put("red", 2);
            put("orange", 3);
            put("yellow", 4);
            put("green", 5);
            put("blue", 6);
            put("violet", 7);
            put("grey", 8);
            put("white", 9);
        }
    };

    int value(String[] colors) {
        return Arrays.stream(colors)
                .map(c -> COLOR_CODES.get(c))
                .limit(2)
                .reduce(0, (a, b) -> a * 10 + b);
    }
}
